scripts/checkout_submodules.py

```diff
@@ -109,10 +109,17 @@ def make_chip_root_safe_directory() -> None:
             ['git', 'config', '--global', '--add', 'safe.directory', CHIP_ROOT])
 
 
-def checkout_modules(modules: list, shallow: bool, force: bool, recursive: bool, jobs: int) -> None:
+def checkout_modules(modules: list, shallow: bool, force: bool, recursive: bool, allow_symlinks: bool, jobs: int) -> None:
     names = ', '.join([module.name for module in modules])
     logging.info(f'Checking out: {names}')
 
+    if allow_symlinks:
+        core_symlinks = subprocess.run(['git', 'config', '--global', '--get', 'core.symlinks'],
+                                       stdout=subprocess.PIPE, text=True).stdout.strip()
+        if core_symlinks == 'false':
+            symlinks_cmd = ['git', 'config', '--global', 'core.symlinks']
+            subprocess.check_call(symlinks_cmd + ['true'])
+
     cmd = ['git', '-C', CHIP_ROOT, 'submodule', '--quiet', 'update', '--init']
```

Comment:
This can be simplified by adding `-c`, `core.symlinks=true` in the update command.